tests/end_to_end.rs

```diff
@@ -1977,3 +1983,139 @@ async fn test_table_time_travel() {
     ];
     assert_batches_eq!(expected, &results);
 }
+
+#[test_case(
+    true;
+    "schema introspected")
+]
+#[test_case(
+    false;
+    "schema declared")
+]
+#[tokio::test]
+async fn test_remote_table_querying(introspect_schema: bool) {
+    let (context, repo) = make_context_with_pg().await;
+
+    // Create a table in our metadata store, and insert some dummy data
+    // TODO: For some reason the TIMESTAMP field fetched by connectorX is a DataType::Date64 instead
+    // of DataType::Timestamp, so that example is omitted for the time being.
+    repo.executor
+        .execute(
+            format!(
+                "CREATE TABLE {}.source_table (a INT, b FLOAT, c VARCHAR, d DATE)",
+                repo.schema_name
+            )
+            .as_str(),
+        )
+        .await
+        .unwrap();
+    repo.executor
+        .execute(
+            format!(
+                "INSERT INTO {}.source_table VALUES \
+            (1, 1.1, 'one', '2022-11-01'),\
+            (2, 2.22, 'two', '2022-11-02'),\
+            (3, 3.333, 'three', '2022-11-03'),\
+            (4, 4.4444, 'four', '2022-11-04')",
```

Comment:
Yeah, arrays and JSON fields don't work because of the limitations of connectorx (in particular no conversion rules for the arrow destination for those fields).

In addition I've found that TIMESTAMPZ field conversion is broken in connectorx—despite there being an [adequate conversion](https://github.com/sfu-db/connector-x/blob/main/connectorx/src/transports/postgres_arrow.rs#L56) definition, the actual implementation is [not there yet](https://github.com/sfu-db/connector-x/blob/main/connectorx/src/destinations/arrow/arrow_assoc.rs#L142-L172).